apiserver/facades/agent/upgrader/upgrader_test.go

```diff
@@ -102,7 +110,13 @@ func (s *upgraderSuite) TestWatchAPIVersion(c *gc.C) {
 func (s *upgraderSuite) TestUpgraderAPIRefusesNonMachineAgent(c *gc.C) {
 	anAuthorizer := s.authorizer
 	anAuthorizer.Tag = names.NewUnitTag("ubuntu/1")
-	anUpgrader, err := upgrader.NewUpgraderAPI(s.State, s.resources, anAuthorizer)
+	anUpgrader, err := upgrader.NewUpgraderAPI(facadetest.Context{
```

Comment:
given the only thing changing between the tests is the authorizer, would it be worth pulling that out into a helper?
Not necessary, just odd to see the same thing over and over again. Its quite a chunk of text.